File: src/main/java/gargoyle/ct/convert/impl/BytesConverter.java
package gargoyle.ct.convert.impl;

import gargoyle.ct.convert.Converter;
import org.jetbrains.annotations.NotNull;

import java.util.Base64;

public class BytesConverter implements Converter<byte[]> {
    @Override
    public String format(byte[] data) {
        return Base64.getEncoder().encodeToString(data);
    }

    @Override
    public byte[] parse(@NotNull String data) {
        return Base64.getDecoder().decode(data);
    }
}

File: src/main/kotlin/gargoyle/ct/convert/impl/BytesConverter.kt
package gargoyle.ct.convert.impl

import gargoyle.ct.convert.Converter
import java.util.Base64

class BytesConverter : Converter<ByteArray> {
    override fun format(data: ByteArray): String {
        return Base64.getEncoder().encodeToString(data)
    }

    override fun parse(data: String): ByteArray {
        return Base64.getDecoder().decode(data)
    }
}
